src/App/Platforms/Android/Autofill/Fido2MakeCredentialUserInterface.cs

```diff
@@ -95,11 +98,35 @@ private async Task<(string CipherId, bool UserVerified)> CreateNewLoginForFido2C
 
         public async Task EnsureUnlockedVaultAsync()
         {
-            if (!await _stateService.IsAuthenticatedAsync() || await _vaultTimeoutService.IsLockedAsync())
+            if (!await _stateService.IsAuthenticatedAsync()
+                ||
+                await _vaultTimeoutService.IsLoggedOutByTimeoutAsync()
+                ||
+                await _vaultTimeoutService.ShouldLogOutByTimeoutAsync())
+            {
+                await NavigateAndWaitForUnlockAsync(Bit.Core.Enums.NavigationTarget.HomeLogin);
+                return;
+            }
+
+            if (!await _vaultTimeoutService.IsLockedAsync())
             {
-                // this should never happen but just in case.
-                throw new InvalidOperationException("Not authed or vault locked");
+                return;
             }
+
+            await NavigateAndWaitForUnlockAsync(Bit.Core.Enums.NavigationTarget.Lock);
+        }
+
+        private async Task NavigateAndWaitForUnlockAsync(Bit.Core.Enums.NavigationTarget navTarget)
+        {
+            _unlockVaultTcs?.TrySetCanceled();
+            _unlockVaultTcs = new TaskCompletionSource<bool>();
+            var messagingService = ServiceContainer.Resolve<IMessagingService>("messagingService");
+            await MainThread.InvokeOnMainThreadAsync(() =>
+            {
+                messagingService?.Send("navigateTo", navTarget);
+            });
```

Comment:
🤔 Is it necessary to send this on the main thread? Additionally, is it necessary to await this?

Reply:
Not needed for sure.